src/components/ThumbnailImage.tsx

```diff
@@ -69,43 +55,41 @@ function calculateThumbnailImageSize(width, height, windowHeight) {
     return {thumbnailWidth: Math.max(40, thumbnailScreenWidth), thumbnailHeight: Math.max(40, thumbnailScreenHeight)};
 }
 
-function ThumbnailImage(props) {
+function ThumbnailImage({previewSourceURL, style, isAuthTokenRequired, imageWidth = 200, imageHeight = 200, shouldDynamicallyResize = true}: ThumbnailImageProps) {
     const {windowHeight} = useWindowDimensions();
-    const initialDimensions = calculateThumbnailImageSize(props.imageWidth, props.imageHeight, windowHeight);
-    const [imageWidth, setImageWidth] = useState(initialDimensions.thumbnailWidth);
-    const [imageHeight, setImageHeight] = useState(initialDimensions.thumbnailHeight);
+    const initialDimensions = calculateThumbnailImageSize(imageWidth, imageHeight, windowHeight);
+    const [currentImageWidth, setCurrentImageWidth] = useState(initialDimensions.thumbnailWidth);
+    const [currentImageHeight, setCurrentImageHeight] = useState(initialDimensions.thumbnailHeight);
 
     /**
      * Update the state with the computed thumbnail sizes.
-     *
-     * @param {{ width: number, height: number }} Params - width and height of the original image.
+     * @param Params - width and height of the original image.
      */
 
     const updateImageSize = useCallback(
-        ({width, height}) => {
+        ({width, height}: {width: number; height: number}) => {
```

Comment:
It would be good to create a separate type for `{width: number; height: number}`

Reply:
moved